File: 0519/12851_숨바꼭질2_백준.py
import sys
sys.stdin = open('12851.txt')
import collections  
dx =[1, -1]

def find_sister():
    deq = collections.deque()
    deq.append((subin, 0))
    limit = 2 * sister
    visit = [-1 for _ in range(limit + 1)]
    visit[subin] = 0
    min_time = float('inf')
    count = 0
    
    while deq:
        witch, time = deq.popleft()
        if time > min_time:
            continue
        if witch == sister:
            if time == min_time:
                count += 1
            else:
                min_time = time
                count += 1
        
        dx = [1, -1, witch]
        
        for w in range(3):
            nx = dx[w] + witch
            if w == 0 or w == 2:
                if nx > limit:
                    continue
            else:
                if nx < 0:
                    continue
            if visit[nx] == -1 or time + 1 <= visit[nx]:
                deq.append((nx, time + 1))
                visit[nx] = time + 1
    
    return min_time, count

    
subin, sister = map(int, input().split())

if subin == sister:
    print(0)
    print(1)
elif subin > sister:
    print(subin - sister)
    print(1)
else:
    min_time, count = find_sister()
    print(min_time)
    print(count)

File: 0519/12851_숨바꼭질2_백준.js
const readline = require('readline')
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
})

input = []
dx = [1, -1]
solution = () => {
    let temp = input[0].split(' ').map((el) => parseInt(el))
    let subin = temp[0]
    let sister = temp[1]
    if (subin === sister) {
        console.log(0)
        console.log(1)
    } else if (subin > sister) {
        console.log(subin - sister)
        console.log(1)
    } else {
        let tteemmpp = findSister(subin, sister)
        console.log(tteemmpp[0])
        console.log(tteemmpp[1])
    }
}

findSister = (subin, sister) => {
    q = []
    q.push([subin, 0])
    limit = 2 * sister
    visit = Array(limit + 1).fill(-1)
    visit[subin] = 0
    minTime = Infinity
    cnt = 0
    while (q.length) {
        let temp = q.shift()
        let witch = temp[0]
        let time = temp[1]
        if (time > minTime) continue;
        if (witch === sister) {
            if (time === minTime) {
                cnt++
            } else {
                minTime = time
                cnt = 1
            }
        }
        let dx = [1, -1, witch]
        for (let w = 0; w < 3; w++) {
            nr = dx[w] + witch
            if (w === 0) {
                if (nr > limit) continue;
            } else if (w === 1) {
                if (nr < 0) continue;
            }
            else {
                if (nr > limit) continue;
            }
            if (visit[nr] == -1 || time + 1 <= visit[nr]) {
                q.push([nr, time + 1])
                visit[nr] = time + 1
            }
        }
    }
    return [minTime, cnt]
}

rl.on('line', line => input.push(line))
.on('close', () => {
    solution()
    process.exit()
})